core/src/test/java/org/apache/iceberg/TestManifestReader.java

```diff
@@ -130,9 +130,6 @@ public void testDataFilePositions() throws IOException {
       long expectedPos = 0L;
       for (DataFile file : reader) {
         assertThat(file.pos()).as("Position should match").isEqualTo(expectedPos);
-        assertThat(((BaseFile) file).get(20))
-            .as("Position from field index should match")
-            .isEqualTo(expectedPos);
```

Comment:
I removed this because the line above tests that `file.pos()` is correct. This assertion is incorrect because it expects the field order not to change.